File: benchmark/benchmark.go
package main

import (
	"fmt"
	"io/ioutil"
	"time"

	"./tinysegmenter"
)

func main() {
	b, err := ioutil.ReadFile("./timemachineu8j.txt")
	if err != nil {
		panic(err)
	}
	sampletext := string(b)

	s := tinysegmenter.NewSegmenter()

	start := time.Now()
	for i := 0; i < 100; i++ {
		s.Segment(sampletext)
	}
	end := time.Now()
	fmt.Println(end.Sub(start))
}

File: benchmark/benchmark.rb
require './tinysegmenter'
require 'benchmark'

text = ''
File.open('timemachineu8j.txt') do |f|
  text = f.read
end

ts = TinySegmenter.new

Benchmark.bm 10 do |r|
  r.report "segment" do
    100.times do
      ts.segment(text)
    end
  end
end
